src/backend/dx12/src/lib.rs

```diff
@@ -146,12 +146,21 @@ pub struct PhysicalDevice {
     private_caps: Capabilities,
     heap_properties: &'static [HeapProperties; NUM_HEAP_PROPERTIES],
     memory_properties: hal::MemoryProperties,
+    // Indicates that there is currently an active logical device.
+    // Opening the same adapter multiple times will return the same D3D12Device again.
+    is_open: Arc<Mutex<bool>>,
```

Comment:
`AtomicBool`?

Reply:
I want to lock the whole creation process and only set `is_open` to `true` once we finished creating the logical device.
Both approaches would work, but a mutex feels safe here.
pseudo code for atomics;
```
if !self.is_open { return Err(); }
self.is_open.set(true);
....
// check device error
if !SUCCESS { self.is_open.set(false); return Err(); }
...
// check queue creation error
if !SUCCESS { self.is_open.set(false); return Err(); }
...
Gpu { device, queues }
```